packages/eslint-plugin/src/rules/restrict-plus-operands.ts

```diff
@@ -1,20 +1,21 @@
 import type { TSESTree } from '@typescript-eslint/utils';
+import * as tsutils from 'tsutils';
 import * as ts from 'typescript';
 
 import * as util from '../util';
 
 type Options = [
   {
-    checkCompoundAssignments?: boolean;
     allowAny?: boolean;
+    allowBoolean?: boolean;
```

Comment:
Similar to the nullish case [the boolean case will create TS errors for numbers, and is only valid when adding with strings](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEYD2A7AzgF3gDwFzwCMkkIQoUBuAWAChbt4BqeARmpoee3c-gHIofdkA).
In the string case it also means you'll be concatenating with `'true' | 'false'` - which doesn't seem desirable in the vast majority of code.

I don't think that this option really makes sense in the context of the rule.